source/apickli/apickli-gherkin.js

```diff
@@ -25,38 +26,38 @@ const callbackWithAssertion = function(callback, assertion) {
     }
 };
 
-module.exports = function() {
-    this.registerHandler('BeforeScenario', function(scenario, callback) {
-        stepContext.scenario = scenario.getName;
-        callback();
-    });
+defineSupportCode(function({Before}) {
+    Before(function(scenarioResult, callback) {
+        // https://github.com/cucumber/cucumber-js/issues/891
+        // stepContext.step = step.getName;
+        // stepContext.scenario = scenario.getName;
 
-    this.registerHandler('BeforeStep', function(step, callback) {
-        stepContext.step = step.getName;
         callback();
     });
+});
 
-    this.Given(/^I set (.*) header to (.*)$/, function(headerName, headerValue, callback) {
+defineSupportCode(function({Given, When, Then}) {
```

Comment:
Feel free to disagree with this statement...

In my mind, support code includes Before and After hooks, as well as environment code such as setting timeouts.
I don't think step definitions are usually defined as support code. (see [here](https://relishapp.com/cucumber/cucumber/docs/writing-support-code) ). 

Must they be inside defineSupportCode to work?